app/controllers/application_controller.rb

```diff
@@ -74,4 +72,13 @@ def evil_parameter_hack!
     # existing controllers as soon as humanly possible.
     params.permit!
   end
+
+  rescue_from CanCan::AccessDenied do |exception|
+    respond_to do |format|
+      format.json { head :forbidden, content_type: 'text/html' }
+      format.html do
+        redirect_back fallback_location: main_app.root_url, alert: exception.message
+      end
+    end
+  end
 end
```

Comment:
This is the centralised error handling I mentioned earlier. It will essentially send the user back to wherever they came from

Reply:
Much better